src/search/product.ts

```diff
@@ -0,0 +1,41 @@
+/**
+ * The structure of a product.
+ */
+export interface Product extends Record<string, any> {
+  /** The name of this product. */
+  title?: string;
+  /** The display price of this product. */
+  price?: number;
+  /** The URL of this product's image. */
+  imageSrc?: string;
+  /** The alternative text for this product's image. */
+  imageAlt?: string;
+  /** The URL for product details page. */
+  productUrl?: string;
+  /** The variants of this product. */
+  variants?: ProductVariants;
+}
+
+/**
+ * The type of a collection of variants.
+ */
+export interface ProductVariants {
+  /** The type of variant rendered. */
+  type: 'color' | 'image' | 'text';
+  /** The list of variants. */
+  items: ProductVariant[];
+}
+
+/**
+ * The structure of a product variant.
+ */
+export interface ProductVariant {
+  /** The variant color. Must be a valid CSS color. */
+  color?: string;
+  /** The URL for the variant thumbnail. */
+  image?: string;
+  /** The label for the variant. */
+  text: string;
```

Comment:
Can we confirm that this is not supposed to be optional? For example, if they provide a color, do they still need to provide a `text` attribute?

Reply:
Looking at the Product component, it looks like it is always used. It is used to either set the `aria-label` for a color or image variant, alt text if it is an image, or the title property of the component (although this isn't actually used anywhere).

Reply:
I think that this is still required because in the case of color, text is being used to set the `aria-label` attribute.